src/lib/alloc.c

```diff
@@ -201,9 +201,30 @@ static void *rmalloc_sys(int zone, int caps, int core, size_t bytes)
 	return ptr;
 }
 
+/* At this point the pointer we have should be unaligned
+ * (it was checked level higher) and be power of 2
+ */
+static void *align_ptr(struct mm_heap *heap, uint32_t alignment,
+		       void *ptr, struct block_hdr *hdr)
+{
+	int mod_align = 0;
+
+	/* Save unaligned ptr to block hdr */
+	hdr->unaligned_ptr = ptr;
+
+	/* If ptr is not already aligned we calculate alignment shift */
+	if ((uintptr_t)ptr % alignment && alignment != 0)
```

Comment:
`if (alignment && (uintptr_t)ptr % alignment)`

`if (alignment)` is the same as `if (alignment != 0)` and it's better to check it first to avoid modulo operation.